src/Listener/CodePatchListener.php

```diff
@@ -36,34 +38,34 @@ public function __construct(CodePatcher $codePatcher)
      */
     public function patch(ExerciseRunnerEvent $event): void
     {
-        $fileName = $event->getInput()->getArgument('program');
+        $files = [$event->getInput()->getArgument('program')];
 
-        if (null === $fileName) {
-            return;
+        $exercise = $event->getExercise();
+        if ($exercise instanceof ProvidesSolution) {
+            $files[] = $exercise->getSolution()->getEntryPoint();
         }
 
-        $this->originalCode = (string) file_get_contents($fileName);
-        file_put_contents(
-            $fileName,
-            $this->codePatcher->patch($event->getExercise(), $this->originalCode)
-        );
+        foreach (array_filter($files) as $fileName) {
+            $this->originalCode[$fileName] = (string) file_get_contents($fileName);
```

Comment:
Does it work if the files are the same name? I think it’s using full paths right so shouldn’t matter?

Reply:
Comes in as full paths from what I can see, so technically shouldn't matter 👍